cpu/stm32/periph/eth.c

```diff
@@ -623,6 +624,11 @@ static int stm32_eth_recv(netdev_t *netdev, void *buf, size_t max_len,
         memcpy(data, rx_curr->buffer_addr, chunk);
         data += chunk;
         remain -= chunk;
+        if (IS_USED(MODULE_PERIPH_PTP) && !remain) {
```

Comment:
Why not move this after the loop?

Reply:
`rx_curr` is after the loop pointing to one item after the DMA descriptor containing. Also, at the end of the loop body the DMA descriptor is handed over to the DMA - at which point we can no longer reliably read its contents.

But its good that you put the finger down to this code. I again forgot to handle the case when one more RX descriptor is there due to the FCS - in which case the timestamp will be placed there.